controllers/snapshot/snapshot_adapter.go

```diff
@@ -82,7 +82,9 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (controller.Operation
 	}
 
 	if integrationTestScenarios != nil {
-		a.logger.Info("Found IntegrationTestScenarios for application",
+		var numberOfIntegrationTestScenario = len(*integrationTestScenarios)
+		a.logger.Info(
+			fmt.Sprintf("Found %v IntegrationTestScenarios for application", numberOfIntegrationTestScenario),
```

Comment:
Can be simplified, also `%d` should be used for numbers
```suggestion
		a.logger.Info(fmt.Sprintf("Found %d IntegrationTestScenarios for application", len(numberOfIntegrationTestScenario)),
```